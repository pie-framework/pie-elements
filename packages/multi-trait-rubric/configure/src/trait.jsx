import React from 'react';
import debug from 'debug';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import { DragSource, DropTarget } from 'react-dnd';

import { withStyles } from '@material-ui/core/styles';
import { color } from '@pie-lib/render-ui';

import {
  Block,
  DragHandleSpace,
  ExpandedInput,
  PrimaryBlock,
  Row,
  SecondaryBlock,
  UnderlinedInput
} from './common';
import { labelPlugins } from './utils';

import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import DragIndicatorIcon from '@material-ui/icons/DragIndicator';

const log = debug('@pie-element:placement-ordering:configure:trait-tile');

const styles = {
  actions: {
    color: color.text()
  },
  controls: {
    display: 'flex',
    justifyContent: 'flex-end',
    position: 'absolute',
    top: '28px',
    right: '8px',
    cursor: 'pointer',
  },
  dragHandle: {
    position: 'absolute',
    top: '80px',
    left: `-${DragHandleSpace}px`,
    cursor: 'move'
  }
};

export class TraitTile extends React.Component {
  state = {};

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.currentPosition !== this.props.currentPosition || (this.secondaryBlock && this.secondaryBlock.scrollLeft !== nextProps.currentPosition)) {
      this.scrollToPosition(nextProps.currentPosition);
    }
  }

  onTraitChanged = (params) => {
    const { trait, onTraitChanged } = this.props;

    if (isEmpty(params)) return;

    Object.keys(params).forEach(key => {
      trait[key] = params[key];
    });

    onTraitChanged(trait);
  };

  onScorePointDescriptorChange = ({ descriptor, value }) => {
    const { trait: { scorePointsDescriptors } } = this.props;

    if (value < 0 || value >= scorePointsDescriptors.length) return;

    scorePointsDescriptors[value] = descriptor;

    this.onTraitChanged({ scorePointsDescriptors });
  };

  handleClick = (event) => this.setState({ anchorEl: event.currentTarget });

  handleClose = () => this.setState({ anchorEl: null });

  scrollToPosition = position => this.secondaryBlock.scrollTo({ left: position });

  openMenu = () => {
    this.props.onTraitRemoved();
    this.handleClose();
  };

  render() {
    const {
      classes,
      connectDragSource,
      connectDropTarget,
      connectDragPreview,
      trait: {
        name,
        standards,
        description,
        scorePointsDescriptors
      },
      scorePointsValues,
      showStandards,
      showDescription,
      enableDragAndDrop
    } = this.props;
    const { anchorEl } = this.state;

    const pluginProps = {
      image: { disabled: true },
      math: { disabled: true }
    };

    return (
      connectDragPreview(
        connectDropTarget(
          <div>
            <Row>
              <PrimaryBlock>
                {enableDragAndDrop ?
                  connectDragSource(
                    <span className={classes.dragHandle}>
                    <DragIndicatorIcon className={classes.actions}/>
                  </span>) : null
              }
              <div className={classes.controls}>
                <IconButton
                  aria-label="more"
                  aria-controls="long-menu"
                  aria-haspopup="true"
                  onClick={this.handleClick}
                >
                  <MoreVertIcon/>
                </IconButton>
                <Menu
                  id="long-menu"
                  anchorEl={anchorEl}
                  keepMounted
                  open={!!anchorEl}
                  onClose={this.handleClose}
                >
                  {[`Remove ${name}`].map((option) => (
                    <MenuItem
                      key={option}
                      onClick={this.openMenu}
                    >
                      {option}
                    </MenuItem>
                  ))}
                </Menu>
              </div>

              <UnderlinedInput
                markup={name}
                onChange={name => this.onTraitChanged({ name })}
                pluginProps={labelPlugins}
                placeholder='Enter Trait'
              />
            </PrimaryBlock>
            <SecondaryBlock
              setRef={ref => {
                this.secondaryBlock = ref;
              }}
            >
              {showStandards && standards && (
                <Block>
                  <ExpandedInput
                    placeholder="Standards"
                    markup={standards.join(',')}
                    onChange={standards => this.onTraitChanged({ standards: standards.split(',') })}
                    pluginProps={pluginProps}
                  />
                </Block>
              )}

                {showDescription && (
                  <Block>
                    <ExpandedInput
                      placeholder="Description"
                      markup={description}
                      onChange={description => this.onTraitChanged({ description })}
                      pluginProps={pluginProps}
                    />
                  </Block>
                )}

                {(scorePointsValues || []).map((scorePointsValue, index) => {
                  const value = scorePointsValues.length - index - 1;
                  let scoreDescriptor;

                  try {
                    scoreDescriptor = scorePointsDescriptors[value] || '';
                  } catch (e) {
                    scoreDescriptor = '';
                  }

                  return (
                    <Block key={`key-key-${index}`}>
                      <ExpandedInput
                        placeholder='Enter Description Here'
                        markup={scoreDescriptor}
                        onChange={descriptor => this.onScorePointDescriptorChange({ descriptor, value })}
                        pluginProps={pluginProps}
                      />
                    </Block>
                  )
                })}
              </SecondaryBlock>
            </Row>
          </div>
        ))
    );
  }
}

TraitTile.propTypes = {
  connectDragSource: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  connectDragPreview: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
  isDragging: PropTypes.bool.isRequired,
  id: PropTypes.any,
  isOver: PropTypes.bool,
  index: PropTypes.number,
  onTraitChanged: PropTypes.func.isRequired,
  onTraitRemoved: PropTypes.func.isRequired,
  trait: PropTypes.shape({
    name: PropTypes.string,
    standards: PropTypes.arrayOf(PropTypes.string),
    scorePointsDescriptors: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
  }),
  scorePointsValues: PropTypes.arrayOf(PropTypes.number),
  showStandards: PropTypes.bool,
  showDescription: PropTypes.bool,
  enableDragAndDrop: PropTypes.bool,
  currentPosition: PropTypes.number,
};

export const StyledTrait = withStyles(styles)(TraitTile);

const NAME = 'trait-config';

const traitSource = {
  beginDrag(props) {
    return {
      name: props.trait.name,
      index: props.index
    };
  }
};

const StyledSource = DragSource(
  NAME,
  traitSource,
  (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    connectDragPreview: connect.dragPreview(),
    isDragging: monitor.isDragging()
  })
)(StyledTrait);

const traitTarget = {
  hover() {
    log('[hover]');
  },
  drop(props, monitor) {
    const item = monitor.getItem();
    log('[drop] item: ', item, 'didDrop?', monitor.didDrop());

    props.onTraitDropped(item, props.index);
  }
};

const StyledSourceAndTarget = DropTarget(
  NAME,
  traitTarget,
  (connect, monitor) => ({
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver()
  })
)(StyledSource);

export default StyledSourceAndTarget;
