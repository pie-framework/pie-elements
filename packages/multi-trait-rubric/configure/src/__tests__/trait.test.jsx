import { shallow } from 'enzyme';
import React from 'react';
import { TraitTile } from '../trait';

const trait = () => ({
  name: 'Presentation',
  standards: [],
  description: '',
  scorePointsDescriptors: [
    'Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. ',
    'Handwriting poor\n' + '\n' + 'Overall appearance is distracting to unacceptable',
    'Handwriting is generally legible\n' + '\n' + 'Overall appearance is acceptable or better',
  ],
});

describe('Trait', () => {
  let w;

  const wrapper = extras => {
    const defaults = {
      classes: {},
      connectDragSource: props => <div>{props}</div>,
      connectDropTarget: props => <div>{props}</div>,
      trait: trait(),
      index: 0,
      scorePointsValues: [0, 1, 2],
      scorePointsLabels: ['A', 'B', 'C'],
      traitLabel: 'Category',
      maxPoints: 2,
      scaleIndex: 0,
      currentPosition: 0,
      showStandards: true,
      showDescription: true,
      ...extras
    };
    return shallow(<TraitTile {...defaults} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();

      expect(w).toMatchSnapshot();
    });

    it('renders without standards', () => {
      w = wrapper({ showStandards: false });

      expect(w).toMatchSnapshot();
    });

    it('renders without description', () => {
      w = wrapper({ description: false });

      expect(w).toMatchSnapshot();
    });

    it('renders with drag and drop', () => {
      w = wrapper({ dragAndDrop: true });

      expect(w).toMatchSnapshot();
    });

    it('renders without score points values', () => {
      w = wrapper({ scorePointsValues: [] });

      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let onScaleChange;
    let onTraitChanged;
    const scrollToPositionSpy = jest.fn();

    beforeEach(() => {
      onTraitChanged = jest.fn();
      onScaleChange = jest.fn();
      w = wrapper({ onScaleChange, onTraitChanged });
    });

    describe('scroll position', () => {
      it ('does not change scroll position when current position prop does not change', () => {
        const wrap = wrapper({ currentPosition: 200 });

        expect(wrap.instance().props.currentPosition).toEqual(200);

        wrap.instance().scrollToPosition = scrollToPositionSpy;
        wrap.setProps({ currentPosition: 200 });

        expect(scrollToPositionSpy).not.toBeCalled();
      });

      it ('changes scroll position when current position prop changes', () => {
        const wrap = wrapper({ currentPosition: 200 });

        expect(wrap.instance().props.currentPosition).toEqual(200);

        wrap.instance().scrollToPosition = scrollToPositionSpy;
        wrap.setProps({ currentPosition: 300 });

        expect(scrollToPositionSpy).toBeCalledWith(300);
      });
    });

    describe('onTraitChanged', () => {
      it('does not call onTraitChanged if params null', () => {
        const { trait } = w.instance().props;

        w.instance().onTraitChanged(null);

        expect(onTraitChanged).not.toBeCalled();
      });

      it('does not call onTraitChanged if params undefined', () => {
        const { trait } = w.instance().props;

        w.instance().onTraitChanged(undefined);

        expect(onTraitChanged).not.toBeCalled();
      });

      it('does not call onTraitChanged if params empty', () => {
        w.instance().onTraitChanged({});

        expect(onTraitChanged).not.toBeCalled();
      });

      it('call onTraitChanged with name', () => {
        const { trait } = w.instance().props;

        w.instance().onTraitChanged({ name: 'New Name' });

        expect(onTraitChanged).toBeCalledWith({
          ...trait,
          name: 'New Name'
        });
      });

      it('call onTraitChanged with standards', () => {
        const { trait } = w.instance().props;

        w.instance().onTraitChanged({ standards: ['a', 'b', 'c'] });

        expect(onTraitChanged).toBeCalledWith({
          ...trait,
          standards: ['a', 'b', 'c']
        });
      });

      it('call onTraitChanged with description', () => {
        const { trait } = w.instance().props;

        w.instance().onTraitChanged({ description: 'New Description' });

        expect(onTraitChanged).toBeCalledWith({
          ...trait,
          description: 'New Description'
        });
      });
    });

    describe('onScorePointDescriptorChange', () => {
      it('does not call onTraitChanged with scorePointsDescriptors', () => {
        w.instance().onScorePointDescriptorChange({ descriptor: 'New Descriptor', value: 10 });

        expect(onTraitChanged).not.toBeCalled();
      });

      it('does not call onTraitChanged with scorePointsDescriptors', () => {
        w.instance().onScorePointDescriptorChange({ descriptor: 'New Descriptor', value: -10 });

        expect(onTraitChanged).not.toBeCalled();
      });

      it('call onTraitChanged with scorePointsDescriptors', () => {
        const { trait } = w.instance().props;

        w.instance().onScorePointDescriptorChange({ descriptor: 'New Descriptor', value: 0 });

        expect(onTraitChanged).toBeCalledWith({
          ...trait,
          scorePointsDescriptors: [
            'New Descriptor',
            ...trait.scorePointsDescriptors.slice(1)
          ]
        });
      });
    });
  });
});
