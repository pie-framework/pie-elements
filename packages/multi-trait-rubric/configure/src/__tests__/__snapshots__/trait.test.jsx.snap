// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trait snapshot renders 1`] = `
<div>
  <div>
    <div>
      <WithStyles(Component)>
        <WithStyles(Component)>
          <div>
            <WithStyles(IconButton)
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="more"
              onClick={[Function]}
            >
              <pure(MoreVertIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              id="long-menu"
              keepMounted={true}
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                key="Remove Trait"
                onClick={[Function]}
              >
                Remove Trait
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
          <WithStyles(Component)
            markup="Presentation"
            onChange={[Function]}
            placeholder="Enter Trait"
            pluginProps={
              Object {
                "image": Object {
                  "disabled": true,
                },
                "math": Object {
                  "disabled": true,
                },
              }
            }
          />
        </WithStyles(Component)>
        <WithStyles(Component)
          setRef={[Function]}
        >
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Standards"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Description"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-0"
          >
            <WithStyles(Component)
              markup="Handwriting is generally legible

Overall appearance is acceptable or better"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-1"
          >
            <WithStyles(Component)
              markup="Handwriting poor

Overall appearance is distracting to unacceptable"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-2"
          >
            <WithStyles(Component)
              markup="Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. "
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
        </WithStyles(Component)>
      </WithStyles(Component)>
    </div>
  </div>
</div>
`;

exports[`Trait snapshot renders with drag and drop 1`] = `
<div>
  <div>
    <div>
      <WithStyles(Component)>
        <WithStyles(Component)>
          <div>
            <WithStyles(IconButton)
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="more"
              onClick={[Function]}
            >
              <pure(MoreVertIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              id="long-menu"
              keepMounted={true}
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                key="Remove Trait"
                onClick={[Function]}
              >
                Remove Trait
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
          <WithStyles(Component)
            markup="Presentation"
            onChange={[Function]}
            placeholder="Enter Trait"
            pluginProps={
              Object {
                "image": Object {
                  "disabled": true,
                },
                "math": Object {
                  "disabled": true,
                },
              }
            }
          />
        </WithStyles(Component)>
        <WithStyles(Component)
          setRef={[Function]}
        >
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Standards"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Description"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-0"
          >
            <WithStyles(Component)
              markup="Handwriting is generally legible

Overall appearance is acceptable or better"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-1"
          >
            <WithStyles(Component)
              markup="Handwriting poor

Overall appearance is distracting to unacceptable"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-2"
          >
            <WithStyles(Component)
              markup="Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. "
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
        </WithStyles(Component)>
      </WithStyles(Component)>
    </div>
  </div>
</div>
`;

exports[`Trait snapshot renders without description 1`] = `
<div>
  <div>
    <div>
      <WithStyles(Component)>
        <WithStyles(Component)>
          <div>
            <WithStyles(IconButton)
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="more"
              onClick={[Function]}
            >
              <pure(MoreVertIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              id="long-menu"
              keepMounted={true}
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                key="Remove Trait"
                onClick={[Function]}
              >
                Remove Trait
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
          <WithStyles(Component)
            markup="Presentation"
            onChange={[Function]}
            placeholder="Enter Trait"
            pluginProps={
              Object {
                "image": Object {
                  "disabled": true,
                },
                "math": Object {
                  "disabled": true,
                },
              }
            }
          />
        </WithStyles(Component)>
        <WithStyles(Component)
          setRef={[Function]}
        >
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Standards"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Description"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-0"
          >
            <WithStyles(Component)
              markup="Handwriting is generally legible

Overall appearance is acceptable or better"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-1"
          >
            <WithStyles(Component)
              markup="Handwriting poor

Overall appearance is distracting to unacceptable"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-2"
          >
            <WithStyles(Component)
              markup="Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. "
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
        </WithStyles(Component)>
      </WithStyles(Component)>
    </div>
  </div>
</div>
`;

exports[`Trait snapshot renders without score points values 1`] = `
<div>
  <div>
    <div>
      <WithStyles(Component)>
        <WithStyles(Component)>
          <div>
            <WithStyles(IconButton)
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="more"
              onClick={[Function]}
            >
              <pure(MoreVertIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              id="long-menu"
              keepMounted={true}
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                key="Remove Trait"
                onClick={[Function]}
              >
                Remove Trait
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
          <WithStyles(Component)
            markup="Presentation"
            onChange={[Function]}
            placeholder="Enter Trait"
            pluginProps={
              Object {
                "image": Object {
                  "disabled": true,
                },
                "math": Object {
                  "disabled": true,
                },
              }
            }
          />
        </WithStyles(Component)>
        <WithStyles(Component)
          setRef={[Function]}
        >
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Standards"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Description"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
        </WithStyles(Component)>
      </WithStyles(Component)>
    </div>
  </div>
</div>
`;

exports[`Trait snapshot renders without standards 1`] = `
<div>
  <div>
    <div>
      <WithStyles(Component)>
        <WithStyles(Component)>
          <div>
            <WithStyles(IconButton)
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="more"
              onClick={[Function]}
            >
              <pure(MoreVertIcon) />
            </WithStyles(IconButton)>
            <WithStyles(Menu)
              id="long-menu"
              keepMounted={true}
              onClose={[Function]}
              open={false}
            >
              <WithStyles(MenuItem)
                key="Remove Trait"
                onClick={[Function]}
              >
                Remove Trait
              </WithStyles(MenuItem)>
            </WithStyles(Menu)>
          </div>
          <WithStyles(Component)
            markup="Presentation"
            onChange={[Function]}
            placeholder="Enter Trait"
            pluginProps={
              Object {
                "image": Object {
                  "disabled": true,
                },
                "math": Object {
                  "disabled": true,
                },
              }
            }
          />
        </WithStyles(Component)>
        <WithStyles(Component)
          setRef={[Function]}
        >
          <WithStyles(Component)>
            <WithStyles(Component)
              markup=""
              onChange={[Function]}
              placeholder="Description"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-0"
          >
            <WithStyles(Component)
              markup="Handwriting is generally legible

Overall appearance is acceptable or better"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-1"
          >
            <WithStyles(Component)
              markup="Handwriting poor

Overall appearance is distracting to unacceptable"
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
          <WithStyles(Component)
            key="key-key-2"
          >
            <WithStyles(Component)
              markup="Handwriting is unreadable, or response is blank, not in English, or too brief to evaluate. "
              onChange={[Function]}
              placeholder="Enter Description Here"
              pluginProps={
                Object {
                  "image": Object {
                    "disabled": true,
                  },
                  "math": Object {
                    "disabled": true,
                  },
                }
              }
            />
          </WithStyles(Component)>
        </WithStyles(Component)>
      </WithStyles(Component)>
    </div>
  </div>
</div>
`;
